v2/src/redirect.ts

```diff
@@ -29,42 +29,49 @@ import {
  */
 export function redirectHandlers(lambdas: lambda.Function[], addLayers: boolean) {
   log.debug(`Wrapping Lambda function handlers with Datadog handler...`);
-  lambdas.forEach((lam) => {
-    const runtime: string = lam.runtime.name;
-    const lambdaRuntime: RuntimeType = runtimeLookup[runtime];
-    if (lambdaRuntime === RuntimeType.JAVA) {
-      lam.addEnvironment(AWS_JAVA_WRAPPER_ENV_VAR, AWS_JAVA_WRAPPER_ENV_VAR_VALUE);
-    } else {
-      const cfnFunction = lam.node.defaultChild as lambda.CfnFunction;
-      if (cfnFunction === undefined) {
-        log.debug("Unable to get Lambda Function handler");
-        return;
-      }
-      const originalHandler = cfnFunction.handler as string;
-      lam.addEnvironment(DD_HANDLER_ENV_VAR, originalHandler);
-      const handler = getDDHandler(lambdaRuntime, addLayers);
-      if (handler === null) {
-        log.debug("Unable to get Datadog handler");
-        return;
-      }
-      cfnFunction.handler = handler;
+
+  for (const l of lambdas) {
+    const runtime: string = l.runtime.name;
+    const runtimeType: RuntimeType = runtimeLookup[runtime];
+
+    if (runtimeType === RuntimeType.JAVA || runtimeType === RuntimeType.DOTNET) {
+      l.addEnvironment(AWS_LAMBDA_EXEC_WRAPPER_ENV_VAR, AWS_LAMBDA_EXEC_WRAPPER);
+      continue;
+    }
+
+    const cfnFuntion = l.node.defaultChild as lambda.CfnFunction;
+    if (cfnFuntion === undefined) {
+      log.debug("Unable to get Lambda Function handler");
```

Comment:
should this be a warn rather than debug log?

Reply:
Not really, I think Container based Lambdas don't have a handler AFAIK.